### PR TITLE
fix: make `paths` a positional argument

### PR DESCRIPTION
**Pull request by klkvr**
## Motivation

Follow-up for #8149. Allows `forge build <dir>` invocations

### CODE REVIEW

File: crates/forge/bin/cmd/build.rs [hunk 1]
```diff
@@ -86,7 +91,7 @@ impl BuildArgs {
 
         // Collect sources to compile if build subdirectories specified.
         let mut files = vec![];
-        if let Some(dirs) = self.args.paths {
+        if let Some(dirs) = self.paths {
             for dir in dirs {
                 files.extend(source_files_iter(dir, MultiCompilerLanguage::FILE_EXTENSIONS));
```
Comment (DaniPopes) [inline crates/forge/bin/cmd/build.rs:96]: dir -> path

and does source_Files_iter work with single files?

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: crates/forge/bin/cmd/build.rs (hunk 1 above)
Comment (klkvr) [reply crates/forge/bin/cmd/build.rs:96]: > and does source_Files_iter work with single files?

yep, from https://docs.rs/walkdir/latest/walkdir/struct.WalkDir.html#method.new:
> If root is a directory, then it is the first item yielded by the iterator. If root is a file, then it is the first and only item yielded by the iterator.

**Review by DaniPopes — APPROVED** [review]

File: crates/cli/src/opts/build/core.rs [hunk 2]
```diff
@@ -126,11 +126,6 @@ pub struct CoreBuildArgs {
     #[serde(skip)]
     pub skip: Option<Vec<SkipBuildFilter>>,
 
-    /// Build source files from specified paths.
-    #[arg(long, short, num_args(0..))]
-    #[serde(skip)]
-    pub paths: Option<Vec<PathBuf>>,
```
Comment (DaniPopes) [inline crates/cli/src/opts/build/core.rs:132]: ah this was in core args, which means it was present in all sub commands, i don't think moving it to pos args is equivalent

Comment (klkvr) [reply crates/cli/src/opts/build/core.rs:132]: it is only used and respected in `BuildArgs::run`, so it belongs to `BuildArgs` anyway

**Comment by klkvr** [comment]
clippy lint looks like a false positive

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**